nav2_system_tests/src/system/test_system_launch.py

```diff
@@ -119,5 +119,26 @@ def main(argv=sys.argv[1:]):
     return lts.run(ls)
 
 
+def main2(argv=sys.argv[1:]):
```

Comment:
Can we rename these functions to something more meaningful, like what these tests are testing (what is the new test testing, exactly?)

Reply:
The new test verify that *makePlan* function return false if an invalid robot start position is given as input.

Reply:
Then why is the return code `true`? You never check any results to know if the correct thing happened?

Reply:
I can't get where you found that exit code. Anyway, you're right, I don't check anywhere that the test result is what I expect (the equivalent of EXPECT_something). It is not clear to me how to do this check in this case, as it is more an integration test than a unit test.

Reply:
yeah but even an integration test needs to check for a given result. The return code is that last statement where you look at `result`. You can add code to the tester node python script to look at more specific items

Reply:
Okay, I'll try to make some changes. Meanwhile, thanks for your feedback.

Reply:
**nav_to_pose_tester_node.py** can already pass/fails depending on input robot position. So I think that a cleaner way to test if a path cannot be created with the given start position if to create another *test_system_launch.py*, which will have different **cmd** arguments. What do you think about it?

Reply:
Alternatively, we could use different configuration files for each test, much like it is now done in *nav2_system_tests/src/system/CMakeLists.txt*.

Reply:
yeah that could be done. Or another input field about "expect failure" so that if you set in a bogus initial pose, you can tell the test you expect everything to fail.

Reply:
You mean something like this `test_action = ExecuteProcess(
        cmd=[os.path.join(os.getenv('TEST_DIR'), os.getenv('TESTER')),
             '-r', '-2.0', '-0.5', '0.0', '2.0', False]` ? 

What about having also initial pose as an ENV argument (`test_action = ExecuteProcess(
        cmd=[os.path.join(os.getenv('TEST_DIR'), os.getenv('TESTER')),
             '-r', os.getenv('TEST_INIT_X'), os.getenv('TEST_INIT_Y'), '0.0', '2.0', False]`), so that it's possible to have a specific test (e.g. *test_invalid_start_position.py*) ?

Reply:
I don't think that would work because you'd still also have to set a flag for whether or not you expect the test to be able to pass or fail, and you need logic within the test itself to take that input and do something with it to fail / succeed the test in the appropriate situation. 

You should just have another launch file with the other starting pose so that you can set the new pose as well as whatever flag you want to send to as an input to that `ExecuteProcess` to denote that you expect this to fail

Reply:
I can definitely create a new launch file to test a wrong initial pose. However, if a new flag is pass to ExecuteProcess I have to modify nav_to_pose_tester_node.py in order to use it, don't I? Because if I have just to check the result of main() function, there's no need to pass no new flag. Sorry, these python tests are driving me crazy 😕

Reply:
Yes, you'd have to modify it to take another argument for whether you expect it to fail or succeed. If succeed, nothing changes. If fail, then when you're running the tests, you should expect the result to be a fail not a success (and then if it does succeed you should fail the test). Just as there are arguments like `-r` you should add a new one for this expect to succeed that can be grabbed in the `main()` function next to the other arguments and then used where it makes sense to in the file. 

E.g. https://github.com/ros-planning/navigation2/blob/6e0db7a37c196cc088428e7b42025351e64882f7/nav2_system_tests/src/system/nav_to_pose_tester_node.py#L406 passed should be treated inverted when you expect it to fail

Reply:
Could you check last commit?

Reply:
Generally looks good!